hibernate-core/src/main/java/org/hibernate/type/ComponentType.java

```diff
@@ -176,9 +176,7 @@ public boolean isEqual(final Object x, final Object y) throws HibernateException
 		if ( x == y ) {
 			return true;
 		}
-		if ( x == null || y == null ) {
```

Comment:
Shouldn't these changes be considered only when this opt-in setting is enabled?  Meaning, to me, I think the setting we add ought to be a gate-keeper between we want the original behavior (default) and we want this new behavior (opt-in).

But as it is, these changed checks for isEqual, isDirty, etc all now happen regardless of whether we opted into the "new behavior"